app/models/manageiq/providers/autosde/inventory/collector/storage_manager.rb

```diff
@@ -26,6 +25,43 @@ def storage_resources
     end
   end
 
+  def addresses
+    @addresses ||= @manager.autosde_client.StorageHostApi.storage_hosts_get.map do |addresses|
```

Comment:
You are calling this API twice. Once here and once in physical_storage_consumers. For performance, it's better to only call it once, and then process the results. 
So I suggest you unify the methods `addresses` and `physical_storage_consumers`.

Reply:
> You are calling this API twice. Once here and once in physical_storage_consumers. For performance, it's better to only call it once, and then process the results.
> So I suggest you unify the methods `addresses` and `physical_storage_consumers`.

I actually think it's clearer if we have a aeparate method for each table. I agree that the same API is called twice but merging it to a single function looks less clear to me.

Reply:
I changed the code so now you only call the api once